### PR TITLE
chore: polish functional tests

### PR DESCRIPTION
**Pull request by Goooler**
Reuse common configurations for Gradle runner.

### CODE REVIEW

**Comment by Goooler** [comment]
CC @KengoTODA

**Comment by Goooler** [comment]
CC @hazendaz

**Review by hazendaz — APPROVED** [review]

**Comment by hazendaz** [comment]
@Goooler Great work!  This is far better now.

File: buildSrc/src/main/kotlin/com/github/spotbugs/com.github.spotbugs.test.gradle.kts [hunk 1]
```diff
@@ -39,7 +39,11 @@ testing {
                 all {
                     testTask.configure {
                         description = "Runs the functional tests."
-                        systemProperty("snom.test.functional.gradle", System.getProperty("snom.test.functional.gradle", gradle.gradleVersion))
+                        var testGradleVersion = providers.gradleProperty("snom.test.functional.gradle").getOrElse("current")
```
Comment (Goooler) [inline buildSrc/src/main/kotlin/com/github/spotbugs/com.github.spotbugs.test.gradle.kts:42]: A typo here, it should be `providers.systemProperty` or use Gradle property, fixing this in gradleVersion

**Pull request merged**